pkg/sql/catalog/errors.go

```diff
@@ -89,10 +90,18 @@ func HasInactiveDescriptorError(err error) bool {
 // found with the given id.
 var ErrDescriptorNotFound = errors.New("descriptor not found")
```

Comment:
Just a musing, but could we upgrade ErrDescriptorNotFound into a struct to discourage using it without providing additional details?

Reply:
Yes we could, and that was my preferred first solution. The issue though is the ~50 or so usages of `errors.Is(err, catalog.ErrDescriptorNotFound)` throughout the codebase, which would no longer work if we changed it to be a struct. Making that change would make the backport slightly more risky.